### DISCUSSION TITLE
Stop Nuxt application if no database connection could be established

### DISCUSSION THREAD

**Discussion by tschni** [nuxt/nuxt#25450] [Questions] [answered]
I would like to realize my backend via the Nuxt server. To establish a database connection directly when starting Nuxt, I created the plugin `database.ts`. If the connection attempt fails (e.g. database not accessible), I want to prevent the application from starting, but stop it immediately.

The plugin currently looks as follows:
```ts
import { consola } from 'consola';

export default defineNitroPlugin(async (nitroApp) => {
  // setup database connection
  try {
    await useDatabase();
  } catch (e) {
    // if the connection to the database has failed,
    // the nuxt application should not be started
    consola.error('Connection to database failed!');
    process.exit(1);
  }
});
```

In the console I also get the message "Connection to database failed!" (strangely enough not in the styling of `consola`), but the process has not terminated. Does anyone have any idea what I need to change?
![image](https://github.com/nuxt/nuxt/assets/58136516/81c696ef-2772-47a8-b62f-92c86adf5474)

Reproduction: https://stackblitz.com/edit/github-vpkeqv?file=server%2Fplugins%2Fdatabase.ts

**Comment by masyoudi** [comment]
I have nuxt app with db connection too, so in my nuxt.config.ts i added hooks like this
```ts
import { dbConnectionTest } from './server/db';

export default defineNuxtConfig({
  hooks: {
    'build:before': async () => {
       try {
         await dbConnectionTest();
       } catch {
         process.exit(1);
       }
    }
  }
});
```

but, this is only works in development / building nuxt.

**Comment by tschni** [comment, marked as answer]
With `process.kill` the process is terminated as desired. This looks as follows:

```ts
import { consola } from 'consola';

export default defineNitroPlugin(async (nitroApp) => {
  // setup database connection
  try {
    await useDatabase();
  } catch (e) {
    // if the connection to the database has failed,
    // the nuxt application should not be started
    consola.error('Connection to database failed!');
    process.kill(process.pid);
  }
});
```
